src/main/webapp/js/finscholar/homepagecontroller/homepagecontroller.js

```diff
@@ -57,13 +48,17 @@ class HomePageController extends PageController {
   }
 
   /**
-   * Do stuff when actions happen.
-   * @param {!JsactionActionFlow} flow Contains the data related to the action
-   *     and more. See actionflow.js.
+   * Renders a view for a scholarship object, which is specified by the uuid.
+   * @public
+   * @param {string} id The uuid of the scholarship to be rendered.
    */
-  doStuff(flow) {
-    // do stuff
-    console.log('doStuff called!');
+  async renderScholarshipPage(id) {
+    try {
+      await this.scholarshipPageHandler_.renderScholarship(id, GoogDom.getElement('content'));
+    } catch(e) {
+      alert(e);
+      console.log(e);
```

Comment:
So I'm making an error template that displays whenever a loading error occurs. So instead of an alert, we would just call the function to render that template in place of where the page would be. I'll let you know when it's done!

Reply:
I think when the main structure is done, I would throw this error to homepagecontroller and handle all errors there.